flake.nix

```diff
@@ -0,0 +1,47 @@
+{
+  description = "Open-source Star Tracker";
+
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs/release-24.05";
+    flake-utils.url = "github:numtide/flake-utils";
+
+    flake-compat = {
+      url = "github:edolstra/flake-compat";
+      flake = false;
+    };
+  };
+
+  outputs = inputs@{ self, nixpkgs, flake-utils, ... }:
+    let
+      overlays = [
+        (final: prev: rec { lost = prev.callPackage ./nix/package.nix { }; })
+      ];
+    in flake-utils.lib.eachDefaultSystem (system:
+      let pkgs = import nixpkgs { inherit overlays system; };
+      in {
+        packages.default = pkgs.lost;
+        packages.lost = pkgs.lost;
+
+        devShells.default = pkgs.mkShell {
+          packages = with pkgs; [
+            gcc
+            clang-tools
+            gnumake
+            bear
+            gdb
+            valgrind
+            man
+            groff
+            imagemagick
```

Comment:
Why do we have imagemagick, graphviz, clang-tools, and bear in here? I realize they may be useful for debugging LOST, does nix have a more idiomatic way of saying "create a development shell for this project, except also install these packages" rather than hardcoding potentially-useful dependencies into this flake file?

Reply:
These all do go into the devShell. The actual binary is built using the `nix/package.nix` where we define nativeBuildInputs (dependencies for building) and buildInputs (runtime dependencies).